authmailbox/message.go

```diff
@@ -0,0 +1,102 @@
+package authmailbox
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const (
+	// MsgMaxSize is the maximum size of a message in bytes.
+	MsgMaxSize = 65536
+)
+
+var (
+	// ErrMessageTooLong is returned when a message exceeds the maximum
+	// allowed length.
+	ErrMessageTooLong = fmt.Errorf("message too long, max %d bytes",
+		MsgMaxSize)
+)
+
+// Message represents a message in the mailbox.
+type Message struct {
+	// ID is the unique identifier for this message, assigned by the
+	// mailbox server.
+	ID uint64
+
+	// ReceiverKey is the intended recipient of the message. This is the
+	// public key of the receiver.
+	ReceiverKey btcec.PublicKey
+
+	// EncryptedPayload is the encrypted message payload. This is the actual
+	// content of the message, encrypted with EICS.
+	EncryptedPayload []byte
+
+	// SenderEphemeralKey is the ephemeral public key of the sender used to
+	// encrypt the message.
+	SenderEphemeralKey btcec.PublicKey
```

Comment:
Mentioned in the proto file, but I don't think we need to store/ref this anywhere.

Reply:
See message above. We need this to decrypt the message. But we can move it to be part of the encrypted payload instead.

Reply:
Ah ok, yeah that makes sense. In my mental model it was "part" of the payload, see this comment: https://github.com/lightninglabs/taproot-assets/pull/1502#discussion_r2110683393.